eduvpn/nm.py

```diff
@@ -138,3 +180,54 @@ def callback(source_object, result):
             logger.debug(source_object.check_connectivity_finish(result))
 
         client.check_connectivity_async(callback=callback)
+
+
+user_dbus_status_callback = None
```

Comment:
I don't really like global variables, unless absolutely necessary. Isn't it possible to store this state in your backend and pass it around?

Reply:
I don't like globals either. In this case the api for add_signal_receiver doesn't have the option to pass user data to the handler so I need to store it somewhere that's accessible from nm.py. I need to register a callback and that one needs to be stored. I can hide the global in a function but that doesn't really solves the issue.

Do you see a better option?

Reply:
In 1.x I did that by having a callback method in the main app that has access to the required resources and add that as a callback to dbus:

https://github.com/eduvpn/python-eduvpn-client/blob/d6c580b17d4acc26bd9335dd226957f81ebc5c42/eduvpn/ui.py#L100

Reply:
I removed the globals.